src/core_functions/scalar/math/numeric.cpp

```diff
@@ -1323,4 +1332,206 @@ ScalarFunctionSet LeastCommonMultipleFun::GetFunctions() {
 	return funcs;
 }
 
+//===--------------------------------------------------------------------===//
+// [divide_by_const]
+//===--------------------------------------------------------------------===//
+
+// This code is inspired by:
+// Lemire et al., Faster Remainder by Direct Computation: Applications to Compilers
+// and Software Libraries, Software: Practice and Experience 49 (6), 2019.
+// https://arxiv.org/abs/1902.01961
+
+static uhugeint_t ComputeM(uint64_t rhs) {
+	return uhugeint_t(0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF) / rhs + 1;
+}
+
+static uint64_t ComputeM(uint32_t rhs) {
+	return UINT64_C(0xFFFFFFFFFFFFFFFF) / rhs + 1;
+}
+
+static uint32_t ComputeM(uint16_t rhs) {
+	return UINT32_C(0xFFFFFFFF) / rhs + 1;
+}
+
+static uint16_t ComputeM(uint8_t rhs) {
+	return UINT32_C(0xFFFF) / rhs + 1;
+}
```

Comment:
Would a template like this make sense so that we can deduplicate code?
```c++
template <class RESULT_TYPE, class RHS_TYPE>
static inline RESULT_TYPE ComputeM(const RHS_TYPE &rhs) {
    return RESULT_TYPE(-1) / rhs + 1;
}
```